src/Orleans.Core/Configuration/Options/GrainDirectoryOptions.cs

```diff
@@ -0,0 +1,66 @@
+
+using System;
+
+namespace Orleans.Hosting
+{
+    public class GrainDirectoryOptions
```

Comment:
I noticed all the option are in orleans.core/configurations. is this by design or ? 

maybe making some exception for some edge case, but as as general thumb of rules, IMO the option class should be in the same project where its component lives.

I will skip the same comment on the option class where I think lived in the wrong place

Reply:
Where options should live is definitely a subject we should consider.  My understanding was that the options that are considered part of our framework (all these options) should be kept in core, and be in the Orleans.Hosting namespace.  Options for extensions and other components should be with those components.  For this PR, this is what I've tried to do.

The reason for the namespace was, if I recall, discoverability.
The reason for keeping them all in core, was that core was all that was needed to configure Orleans before, so moving these around could introduce new dependencies.

What should be mostly clear from the state of the code before this PR is that the pattern I've followed here conformed to the pre-existing pattern and revisiting that pattern was not part of this work.

Reply:
I think either `Orleans` or `Orleans.Configuration`, but I'm not sure. What matter most is discoverability